src/order/ideal.lean

```diff
@@ -151,6 +168,22 @@ end⟩
 @[mk_iff] class is_maximal (I : ideal P) extends is_proper I : Prop :=
 (maximal_proper : ∀ ⦃J : ideal P⦄, I < J → J.carrier = set.univ)
 
+lemma ideal_Inter_nonempty.exists_all_mem (hP : ideal_Inter_nonempty P) :
```

Comment:
Shouldn't this be a typeclass argument?
```suggestion
lemma ideal_Inter_nonempty.exists_all_mem [ideal_Inter_nonempty P] :
```

Reply:
I tried this but this doesn't work easily with lemmas that use this form (for example, changing the proof of `ideal_Inter_nonempty.ideal_bInter_nonempty` to use this just doesn't work for some reason), so I think it is better to not include it.

Reply:
It's possible to make Eric's suggestion work. 

```
lemma ideal_Inter_nonempty_iff :
  ideal_Inter_nonempty P ↔ ∃ a : P, ∀ I : ideal P, a ∈ I :=
⟨λ h, by exactI ideal_Inter_nonempty.exists_all_mem, ideal_Inter_nonempty_of_exists_all_mem⟩
```

and then later on (three times) use `obtain` e.g.

```
instance ideal_Inter_nonempty.ideal_inter_nonempty : ideal_inter_nonempty P :=
{ inter_nonempty := λ _ _, begin
    obtain ⟨a, ha⟩ : ∃ a : P, _ := ideal_Inter_nonempty.exists_all_mem,
    exact ⟨a, ha _, ha _⟩,
  end }
```